bin/logstash.bat

```diff
@@ -47,5 +50,34 @@ set JAVA_OPTS=%LS_JAVA_OPTS%
 
 rem jruby launcher will pickup JAVA_OPTS set above to set the JVM options before launching jruby
 %JRUBY_BIN% "%LS_HOME%\lib\bootstrap\environment.rb" "logstash\runner.rb" %*
+goto end
+
+:version
+rem make the script work without LS_HOME variable set
+if "%LS_HOME%" == "" (
+	for %%a in ("%~dp0..") do set "LS_HOME=%%~fa"
+)
+
+set "LOGSTASH_VERSION_FILE1=%LS_HOME%\logstash-core\versions-gem-copy.yml"
+set "LOGSTASH_VERSION_FILE2=%LS_HOME%\versions.yml"
+
+set "LOGSTASH_VERSION=Version not detected"
+if exist !LOGSTASH_VERSION_FILE1! (
+	rem this file is present in zip, deb and rpm artifacts and after bundle install
+	rem but might not be for a git checkout type install
+	for /F "tokens=1,2 delims=: " %%a in (!LOGSTASH_VERSION_FILE1!) do (
+		if "%%a"=="logstash" set LOGSTASH_VERSION=%%b
+	)
+) else (
+	if exist !LOGSTASH_VERSION_FILE2! (
+		rem this file is present for a git checkout type install
+		rem but its not in zip, deb and rpm artifacts (and in integration tests)
+		for /F "tokens=1,2 delims=: " %%a in (!LOGSTASH_VERSION_FILE2!) do (
+			if "%%a"=="logstash" set LOGSTASH_VERSION=%%b
+		)
+	)
+)	
+echo "logstash !LOGSTASH_VERSION!"
 
```

Comment:
++ this seems consistent with the bash version logic. will manually test shortly. 👍